ci/build.py

```diff
@@ -50,8 +50,8 @@ def tox(tox_args, build_dir):
     if conf.exists():
         conf.unlink()
 
-    # Set a reasonable size limit
-    sp.check_call('ccache --max-size=500M', shell=True)
+    # Set a reasonable size limit, and compare compiler based on content, not mtime
+    sp.check_call('ccache --max-size=500M -o compiler_check=content', shell=True)
```

Comment:
Don't we already enable that in line 73? That said, the code seems to have changed a bit since I last checked... this args.cmake is new to me.

Reply:
Oh right! I have only looked at the output of `ccache -p` in the ci (which is before line 73) and noticed this setting. Do you have an idea what else could cause the 100% cache miss rate?

Reply:
Without having looked at it more closely, I'd suggest trying to disable the hash_dir option. This is a common culprit in environments where every build happens in a randomized path. No idea if that applies to GHA, though.

Reply:
According to the log, the directory is always `/__w/herbstluftwm/herbstluftwm/build` (not sure whether the log is de-randomized). I've just set `hash_dir=false`, so let us see what happens.